system/ee/ExpressionEngine/Addons/channel/libraries/channel_form/Channel_form_lib.php

```diff
@@ -2705,7 +2705,12 @@ public function initialize($reinitialize = false)
         );
 
         $this->custom_fields = array();
-        $this->custom_option_fields = array();
+
+        if ($custom_option_fields = ee('Model')->get('Config')->filter('key', 'custom_option_fields')->filter('site_id', $this->site_id)->first()) {
```

Comment:
Is there a reason why this is being restricted to current site ID and not just taken from general system configuration?